spec/eofv0_verkle.md

```diff
@@ -105,6 +105,158 @@ The same as above except encode the values as 6-bit numbers
 (minimum number of bits needed for encoding `32`).
 Such encoding lowers the size overhead from 3.1% to 2.3%.
 
+### Encode only invalid jumpdests (dense encoding)
+
+Alternate option is instead of encoding all valid `JUMPDEST` locations, to only encode invalid ones.
+By invalid `JUMPDEST` we mean a `0x5b` byte in any pushdata.
+
+This is beneficial if our assumption is correct that most contracts only contain a limited number
+of offending cases. Our initial analysis suggests this is the case, e.g. Uniswap router has 9 cases,
+one of the Arbitrum validator contracts has 6 cases.
+
+Since Solidity contracts have a trailing metadata, which contains a Keccak-256 (32-byte) hash of the
+source, there is a 12% probability ($1 - (255/256)^{32}$) that at least one of the bytes of the hash
+will contain the `0x5b` value, which gives our minimum probability of having at least one invalid
+`JUMPDEST` in the contract.
+
+Let's create a map of `invalid_jumpdests[chunk_no] = first_instruction_offset`. We can densely encode this
+map using techniques similar to *run-length encoding* to skip distances and delta-encode offsets.
+This map is always fully loaded prior to execution, and so it is important to ensure the encoded
+version is as dense as possible (without sacrificing on complexity).
+
+In *scheme 1*, for each entry in `invalid_jumpdests`:
+- 1-bit mode (`skip`, `value`)
+- For skip-mode:
+  - 10-bit number of chunks to skip
+- For value-mode:
+  - 6-bit `first_instruction_offset`
```

Comment:
one question that came up: there can be more than one entry per chunk, if there are more than one `PUSHn` in the chunk? Why not store just the overflowing one?

Reply:
I'm not sure what "the overflowing one" means.
In the current version for a chunk what has any number of invalid jumps we store _first instruction offset_ as in the "vanilla" verkle EIP. This requires to perform the jumpdest analysis on the chunk (as in the "vanilla" verkle).

There are some alternatives to _first instruction offest_ but we currently aim for storing single number per chunk because this really binds the worst case.